frontend/src/containers/MembersPage/MemberList.tsx

```diff
@@ -1,136 +1,49 @@
-            <Text fontSize={"2xl"} fontWeight={"semibold"}>
-              John Doe
-            </Text>
-          </Link>
-        </Box>
-        <Box
-          border={"1px"}
-          borderRadius={"16px"}
-          p={"6"}
-          w={"full"}
-          backgroundColor={"gray.700"}
-          cursor={"pointer"}
-        >
-          <Link href={"/payments/1"}>
-            <Text fontSize={"2xl"} fontWeight={"semibold"}>
-              John Doe
-            </Text>
-          </Link>
-        </Box>
-        <Box
-          border={"1px"}
-          borderRadius={"16px"}
-          p={"6"}
-          w={"full"}
-          backgroundColor={"gray.700"}
-          cursor={"pointer"}
-        >
-          <Link href={"/payments/1"}>
-            <Text fontSize={"2xl"} fontWeight={"semibold"}>
-              John Doe
-            </Text>
-          </Link>
-        </Box>
-        <Box
-          border={"1px"}
-          borderRadius={"16px"}
-          p={"6"}
-          w={"full"}
-          backgroundColor={"gray.700"}
-          cursor={"pointer"}
-        >
-          <Link href={"/payments/1"}>
-            <Text fontSize={"2xl"} fontWeight={"semibold"}>
-              John Doe
-            </Text>
-          </Link>
-        </Box>
-        <Box
-          border={"1px"}
-          borderRadius={"16px"}
-          p={"6"}
-          w={"full"}
-          backgroundColor={"gray.700"}
-          cursor={"pointer"}
-        >
-          <Link href={"/payments/1"}>
-            <Text fontSize={"2xl"} fontWeight={"semibold"}>
-              John Doe
-            </Text>
-          </Link>
-        </Box>
-        <Box
-          border={"1px"}
-          borderRadius={"16px"}
-          p={"6"}
-          w={"full"}
-          backgroundColor={"gray.700"}
-          cursor={"pointer"}
-        >
-          <Link href={"/payments/1"}>
-            <Text fontSize={"2xl"} fontWeight={"semibold"}>
-              John Doe
-            </Text>
-          </Link>
-        </Box>
+        {match(query)
+          .with({ status: "success" }, (query) =>
+            query.data.data.map((member) => {
+              return (
+                <Box
+                  key={member.id}
+                  border={"1px"}
+                  borderRadius={"16px"}
+                  p={"6"}
+                  w={"full"}
+                  backgroundColor={"gray.700"}
+                  cursor={"pointer"}
+                >
+                  <Link href={`/payments/${member.id}`}>
+                    <Text fontSize={"2xl"} fontWeight={"semibold"}>
+                      {member.name}
+                    </Text>
+                  </Link>
+                </Box>
+              );
+            })
+          )
+          .with({ status: "loading" }, () => (
+            <Spinner color="blue.500" size="xl" thickness="4px" />
+          ))
+          .with({ status: "error" }, (query) => (
+            <Text>{query.data?.message}</Text>
```

Comment:
ini kenapa possibly undefined yaa? jadi pada saat nge handle status error data nya undefined dan dapat message error gini dari react-query nya: `Cannot read properties of undefined (reading 'status')`
apakah karena di json-server nya hanya ngehandle success aja?

Reply:
ini kayaknya belum bisa dapat error message dari BE nya. Perlu diperbaiki di json schemanya. Kayaknya dikasih error message general dulu ? misalnya `Something went wrong` ?

Reply:
error general berarti di komponent nya yaa?
```
.with({ status: "error" }, (query) => (
            <Text>Something went wrong</Text>
          ))
```

Reply:
iya betul